config/orte_setup_debugger_flags.m4

```diff
@@ -24,6 +24,21 @@ dnl
 dnl $HEADER$
 dnl
 
+dnl Check to see if specific CFLAGS work
+dnl $1: compiler flags to check
+dnl $2: Action if the flags work
```

Comment:
Curiosity question: did you not like the 3-var variant that was proposed?  (i.e., `$3` = action if flags do not work).  I ask because this is a common mechanism (some test, and then "do this if it works" / "do this if it doesn't work").  I'm not strictly tied to this, but we do use that convention all throughout our m4.

Such an approach lets you have a compact form down below:

```
_ORTE_SETUP_DEBUGGER_FLAGS_TRY_CFLAGS([-fasynchronous-unwind-tables],
        [MPIR_UNWIND_CFLAGS=-fasynchronous-unwind-tables],
        [_ORTE_SETUP_DEBUGGER_FLAGS_TRY_CFLAGS([-Meh_frame -Mframe],
            [MPIR_UNWIND_CFLAGS="-Meh_frame -Mframe"],
            [MPIR_UNWIND_CFLAGS=-g])
        ])
```

Reply:
I tried this one a couple of different ways, and always ended up with a syntax error in the generated configure script. I assumed that nesting functions wasn't supported. 

This also seems to suggest the same: http://www.slac.stanford.edu/comp/unix/package/rtems/doc/html/autoconf/autoconf.info.Multiple_Cases.html

Reply:
m4 is a fickle beast.  It's like a bully on the playground: if you give it all your lunch money, it might not beat you up.  😄 

m4 nesting is fully supported.  You just have to be careful about side effects -- remembering that m4 macros are *expanded*, not *called*.  My original suggestion typed here in Github wasn't quite correct.  I pushed a commit to this PR that seems to work.  What do you think?

I also rebased this PR on top of the current v3.1.x branch to get rid of the orte_rml_ofi warnings during `autogen.pl` (we fixed those in a different PR).

Reply:
Thanks for the help, this seems much better. I've rebased it onto master and made a new PR: https://github.com/open-mpi/ompi/pull/6527